src/main/scala/com/advancedtelematic/campaigner/data/DataType.scala

```diff
@@ -121,6 +122,20 @@ object DataType {
 
   final case class UpdateCampaign(name: String, metadata: Option[Seq[CreateCampaignMetadata]] = None)
 
+  final case class SearchCampaignParams(status: Option[CampaignStatus],
+                                        nameContains: Option[String],
+                                        withErrors: Option[Boolean],
+                                        sortBy: SortBy,
+                                        offset: Long,
+                                        limit: Long,
+                                       ) {
+    if (withErrors.isDefined) {
```

Comment:
Are we handling `IllegalArgumentException` somewhere?

Reply:
Yes, Akka Http takes care of it automatically for this "case class extraction" and returns a 400. See https://doc.akka.io/docs/akka-http/current/routing-dsl/case-class-extraction.html#case-class-validation.